### ISSUE TITLE
Output always says "cert not due for renewal", "No renewals were attempted" even when it does renew

### ISSUE THREAD

**Issue by php4fan** [certbot/certbot#7826]
## My operating system is (include version):

Debian 10.2

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):

I don't remember. How can I find out?

## I ran this command and it produced this output:

On 2020-02-23, this command was run by a cronjob and I received the output via email:

```
$ certbot renew

Saving debug log to /var/log/letsencrypt/letsencrypt.log

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Processing /etc/letsencrypt/renewal/*****.conf
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Cert not yet due for renewal

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

The following certs are not due for renewal yet:
  /etc/letsencrypt/live/******.com/fullchain.pem expires on 2020-03-24 (skipped)
No renewals were attempted.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

So, it says that the certificate is not due for renewal and that no renewal was attempted.

Now, on 2020-02-29 I have run the same command from the CLI manually and got this output:
```
$ certbot renew

Saving debug log to /var/log/letsencrypt/letsencrypt.log

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Processing /etc/letsencrypt/renewal/*****.conf
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Cert not yet due for renewal

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

The following certs are not due for renewal yet:
  /etc/letsencrypt/live/******.com/fullchain.pem expires on 2020-05-23 (skipped)
No renewals were attempted.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

```

## Certbot's behavior differed from what I expected because:

Note that the expiration date has changed. That is, on feb 23 it was supposedly not due for renewal and it was supposedly not renewed, yet on feb 29 automagically the expiration date has changed and hence it is still not due for renewal. This makes no sense. In one of the two instances, the output that says "no renewals were attempted" is bullshit. In either one of the two instances, the certificate was renewed.

By inspecting /var/log/letsencrypt/letsencrypt.log, it appears that when it ran on feb 23, it did in fact renew the certificate:

```
2020-02-23 22:15:00,819:INFO:certbot.main:Renewing an existing certificate
```

And indeed by visiting my website in Chrome and looking at the certificate information, I see it was issued on feb 23 and expires on may 24.

So, the script systematically outputs "cert not due for renewal" and "no renewals were attempted" regardless of whether or not that is the case.

It works correctly in detecting when the certificate needs to be renewed and in renewing it, but if the output is not reliable it's like having no output. Actually, it's worse than having no output, because you think that you are getting information when you are not.

This, in addition to the fact that when you create a new certificate with some of the same domains but some more domains added to it, you keep receiving expiration emails from LetsEncrypt about the old certificate, makes everything a mess where you never know what the **** is going on and end up having to check everything manually. Which defeats the purpose of having notification emails.

**Comment by alexzorin** [comment]
I think you might have two different cronjobs setup.

If you had installed Certbot from the Debian Buster repositories (having followed the instructions from https://certbot.eff.org/lets-encrypt/debianbuster-other), that package comes with its own cronjob (or systemd timer, if you have systemd installed):

    /etc/cron.d/certbot
    /lib/systemd/system/certbot.{service,timer}

Do those exist on your system?

That cronjob, which runs twice a day, **does not generate any non-error output** (because it runs `certbot renew` with the `--quiet` option).

If you are receiving emails with `No renewals were attempted`, that might suggest that you've got a redundant cron jobsetup.

I would guess the true timeline is that the package's cronjob renewed the certificate (quietly), in between runs of your own (non-`--quiet`) cronjob.

>you keep receiving expiration emails from LetsEncrypt about the old certificate, makes everything a mess where you never know what the **** is going on and end up having to check everything manually

These aren't part of the Certbot project, but come from the ACME CA itself. Feedback welcome here: https://community.letsencrypt.org/

**Comment by schoen** [comment]
I agree with @alexzorin's interpretation here. Maybe the remedy for this is a documentation or packaging change that will help people avoid duplicate renewal cronjobs?

**Comment by php4fan** [comment]
Indeed,  I had the automatically installed cron at /etc/cron.d/certbot and the one I created myself in /etc/crontab.

I'm pretty sure I followed installation instructions somewhere. It's possible that they didn't explicitly mention to create a cronjob, I may have got the idea that I should do that because I had had a previous installation of an older version on another machine where (and of that I'm certain) I had had to create the cronjob because the instructions said so. So it's possible that I jumped to the conclusion that I needed to do so and that I just looked for the command to renew and copied that into the crontab. I certainly didn't see anything in the installation instructions or docs that mentioned the existence of a cronjob coming with the package, but I might have missed that.


Actually I now see this:
> The command to renew certbot is installed in one of the following locations:
>
> /etc/crontab/
> /etc/cron.*/*
> systemctl list-timers

I think the phrasing is pretty unclear.
